scripts/roll_release.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+#
+# Distributed Linear Algebra with Future (DLAF)
+#
+# Copyright (c) 2018-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+# This script tags a release locally and creates a release on GitHub. It relies
+# on the hub command line tool (https://hub.github.com/).
+
+set -o errexit
+
+VERSION_MAJOR=$(sed -n 's/project(DLAF VERSION \([0-9]\+\)\.[0-9]\+\.[0-9]\+)/\1/p' CMakeLists.txt)
+VERSION_MINOR=$(sed -n 's/project(DLAF VERSION [0-9]\+\.\([0-9]\+\)\.[0-9]\+)/\1/p' CMakeLists.txt)
+VERSION_PATCH=$(sed -n 's/project(DLAF VERSION [0-9]\+\.[0-9]\+\.\([0-9]\+\))/\1/p' CMakeLists.txt)
+VERSION_FULL="${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}"
+VERSION_FULL_TAG="v${VERSION_MAJOR}.${VERSION_MINOR}.${VERSION_PATCH}"
+VERSION_TITLE="DLA-Future ${VERSION_FULL}"
+CURRENT_BRANCH="$(git rev-parse --abbrev-ref HEAD)"
+RELEASE_BRANCH="release-${VERSION_MAJOR}.${VERSION_MINOR}"
```

Comment:
Maybe a note about the possibility to set RELEASE_BRANCH="master"?
I would avoid to create extra branches if unnecessary.

Reply:
My arguments for always requiring an explicit `release-*` branch are:
- it's explicit and requires working on the correct branch/commit
- the release script can check that you're on the correct branch
- work can continue independently on master while a release is being made
- there is no special case

I understand the argument for simply using `master` as well.

Reply:
But it increases the chances of having diverging branches...

Reply:
Sure, that is true as well. In the end I don't care much either way as long as the release procedure helps us all make more reliable releases. That's after all what you have been doing so far so it makes sense to start there. Using a release branch is just my personal preference. I'll relax the checks in the script a bit to allow `master` as well.

Reply:
Update the release procedure and script in https://github.com/eth-cscs/DLA-Future/pull/1127/commits/7ca6fcaf034c83035af289de6d34e5dbf2bea8f3 and https://github.com/eth-cscs/DLA-Future/pull/1127/commits/8328e9acc76639c19604a7790575cbae3fd5d0a4.